extensions/azure/storage/AzureDataLakeStorage.cpp

```diff
@@ -72,4 +77,62 @@ std::optional<uint64_t> AzureDataLakeStorage::fetchFile(const FetchAzureDataLake
   }
 }
 
+bool AzureDataLakeStorage::matchesPathFilter(const std::string& base_directory, const std::string& path_filter, std::string path) {
+  if (path_filter.empty()) {
+    return true;
+  }
+
+  if (!base_directory.empty()) {
+    gsl_Expects(minifi::utils::StringUtils::startsWith(path, base_directory));
+    path = path.size() == base_directory.size() ? "" : path.substr(base_directory.size() + 1);
+  }
+
+  std::regex pattern(path_filter);
+  return std::regex_match(path, pattern);
+}
+
+bool AzureDataLakeStorage::matchesFileFilter(const std::string& file_filter, const std::string& filename) {
```

Comment:
Clang-Tidy: Method 'matchesFileFilter' can be made static

In general, if it can be static and private, it can also be hidden in the implementation file in an anonymous namespace. I prefer it that way, but not everyone agrees with this.

Consider using `std::string_view` instead of `const std::string&` to pass strings.

Reply:
Updated ef0172d119c472a1c91a10aecbe531a112ade2bf